### PR TITLE
Converting models to Kotlin.

### PR DESCRIPTION
**Pull request by reul**
Those models were converted to Kotlin using the Android Studio convert to kotlin action:

- Account
- AccountType
- BaseModel
- Book
- Commodity
- Money
- PeriodType
- Price
- Recurrence
- ScheduledAction
- ~Transaction~
- TransactionType.

### CODE REVIEW

**Comment by reul** [comment]
I've started converting the models to Kotlin. This PR includes only changes that won't require major refactorings only.

I'm trying to keep PRs small, so they are not too tedious to review. A few others are going to follow.

**Comment by reul** [comment]
@djbrown: if we're merging this, please squash merge. I'm trying to avoid to rebase + force-push because sometimes github does funny stuff with the PR.

File: app/src/main/kotlin/org/gnucash/android/model/TransactionType.kt [hunk 1]
```diff
@@ -13,33 +13,34 @@
...
-
-package org.gnucash.android.model;
+package org.gnucash.android.model
 
 /**
  * Type of transaction, a credit or a debit
  *
  * @author Ngewi Fet <ngewif@gmail.com>
- * @author Jesse Shieh <jesse.shieh.pub@gmail.com>
+ * @author Jesse Shieh <jesse.shieh.pub></jesse.shieh.pub>@gmail.com>
```
Comment (djbrown) [inline app/src/main/kotlin/org/gnucash/android/model/TransactionType.kt:22]: broken email - probably falsely identified as html tag

File: app/src/main/kotlin/org/gnucash/android/model/ScheduledAction.kt [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+    /**
+     * Returns the total number of planned occurrences of this scheduled action.
+     *
+     * @return Total number of planned occurrences of this action
+     */
+    /**
+     * Sets the number of occurences of this action
+     *
+     * @param plannedExecutions Number of occurences
+     */
```
Comment (djbrown) [inline app/src/main/kotlin/org/gnucash/android/model/ScheduledAction.kt:105]: too much javadoc

File: app/src/main/kotlin/org/gnucash/android/model/ScheduledAction.kt [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+    /**
+     * Returns how many times this scheduled action has already been executed
+     *
+     * @return Number of times this action has been executed
+     */
+    /**
+     * Sets the number of times this scheduled action has been executed
+     *
+     * @param executionCount Number of executions
+     */
```
Comment (djbrown) [inline app/src/main/kotlin/org/gnucash/android/model/ScheduledAction.kt:119]: too much javadoc

**Review by djbrown — CHANGES_REQUESTED** [review]
conversion of the "production" code itself looks good to me 👍🏾 
but some javadoc comments got mixed up (see comments)
also I'm not sure wether `@param` can be used on attributes (I'd hope the IDE would check that).

File: app/src/main/kotlin/org/gnucash/android/model/Recurrence.kt [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+        }
+
+//        =======================================================================
+//This section complies with the formal rules, but the betterpickers library doesn't like/need it
+//        SimpleDateFormat startDateFormat = new SimpleDateFormat("'TZID'=zzzz':'yyyyMMdd'T'HHmmss", Locale.US);
+//        ruleBuilder.append("DTSTART;");
+//        ruleBuilder.append(startDateFormat.format(new Date(mStartDate)));
+//            ruleBuilder.append("\n");
+//        ruleBuilder.append("RRULE:");
+//        ========================================================================
```
Comment (djbrown) [inline app/src/main/kotlin/org/gnucash/android/model/Recurrence.kt:128]: duplicate commented out code block (see below L144-L151)

**Review by djbrown — CHANGES_REQUESTED** [review]
again overall LGTM, just two duplicate comment blocks

File: app/src/main/kotlin/org/gnucash/android/model/Recurrence.kt (hunk 4 above)
Comment (reul) [reply app/src/main/kotlin/org/gnucash/android/model/Recurrence.kt]: Sorry! I didn't even noticed those. They should be gone now.

**Review by djbrown — APPROVED** [review]

**Comment by djbrown** [comment]
@reul the `Transaction` model wasn't converted, so I updated the PR description to reflect that.
I guess this class will be converted in an upcoming PR. Thank you so far 😊

**Comment by reul** [comment]
Thanks. I'm going to send a PR with the remaining ones later today.

**Pull request merged**